boards/arm/lpcxpresso55s36/lpcxpresso55s36_defconfig

```diff
@@ -15,5 +15,6 @@ CONFIG_GPIO=y
 CONFIG_PINCTRL=y
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=96000000
 CONFIG_ARM_MPU=y
+CONFIG_REGULATOR=y
```

Comment:
I don't think we want to unconditionally enable the VREF for the LPC55S36, since it increases power consumption.

Could we instead enable the regulator API when the `CONFIG_ADC=y` in the board's defconfig?

Reply:
This, together with DT status ok will enable the driver build on any board build. If not needed for basic functioning of the board, please remove it would not fall into the category of minimal peripheral.